spec/controllers/devise/two_factor_authentication_controller_spec.rb

```diff
@@ -1,51 +1,58 @@
 require 'rails_helper'
 
 describe Devise::TwoFactorAuthenticationController, devise: true do
-  describe 'update' do
+  describe '#update' do
     let(:user) { create(:user, :signed_up) }
 
+    context 'when resource is fully authenticated' do
+      it 'redirects to the dashboard' do
+        sign_in user
+        get :new
+
+        expect(response).to redirect_to dashboard_index_path
+      end
+    end
+
```

Comment:
Gotcha. In that case, I don't think we need this test because a user cannot make a patch request directly without visiting the show page first.